.github/workflows/github_pages.yml

```diff
@@ -0,0 +1,64 @@
+name: Deploy to GitHub Pages
+on:
+  workflow_call:
+    inputs:
+      settings:
+        type: string
+        default: "publishconf.py"
+      output-path:
+        type: string
+        default: "output"
+      cache:
+        type: boolean
+      cache-path:
+        type: string
+        default: "cache"
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - name: Configure GitHub Pages
+        id: pages
+        uses: actions/configure-pages@v3
+      - name: Install requirements
+        run: pip install "pelican[markdown]"
+      - name: Print Pelican settings
+        run: pelican --print-settings > .pelican_settings.json
+        shell: bash
+      - name: Cache
+        uses: actions/cache@v3
+        with:
+          key: ${{ runner.os }}-${{ hashFiles('.pelican_settings.json') }}
+          path: ${{ inputs.cache-path }}
```

Comment:
I'm using a hash of the output of `pelican --settings` as the cache key, so if any of the user's Pelican settings have changed that should invalidate the cache, even if the user is using something other than `publishconf.py` for their settings file or if their settings file imports other files etc. See https://github.com/actions/cache#creating-a-cache-key

Reply:
Hmm, I'm not sure if this is correct. GitHub Actions caches are immutable, they can't be updated. So we need to make sure that whenever the input contents change the cache key changes so that we create a new cache. I think the cache key needs to include hashes of everything that goes into creating the site, including the `content` directory, the settings file and any other files that it imports or depends on, and anything else that affects the result of rendering the site.

Reply:
> something other than publishconf.py

Not true. Any `pelican` command that does not specify the config file will look for and load `pelicanconf.py`. So, `pelican --print-settings` will dump settings defined in `pelicanconf.py` if it exists or just default values if it doesn't.

Reply:
:facepalm: Yes, of course. I think the cache key should just be based on the git commit SHA then, but for now I've removed it from this PR: it can be done in a separate PR since it's a tricky problem on its own